File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        if len(nums) <= 1:
            return n
        
        left = 0
        for right in range(0, n):
            if nums[right] != nums[left]:
                left += 1
                nums[left] = nums[right]
        
        return left + 1

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int writer = 0;

        for(int reader = 0; reader < nums.length; reader++){
            if(nums[reader] != nums[writer]){
                writer += 1;
                nums[writer] = nums[reader];
            }
        }

        return writer + 1;
    }
}